src/web/mod.rs

```diff
@@ -142,22 +142,37 @@ impl Handler for CratesfyiHandler {
         // return 404 if none of them return Ok
         self.shared_resource_handler
             .handle(req)
-            .or_else(|e| self.router_handler.handle(req).or(Err(e)))
+            .or_else(|e| {
+                if e.response.status == Some(status::NotFound) {
+                    self.router_handler.handle(req)
+                } else {
+                    Err(e)
+                }
+            })
```

Comment:
Suggestions on ways to rewrite this would be appreciated, seems like it needs some sort of conditional `or_else_if(|e| e.response.status == Some(status::NotFound), |e| self.router_handler.handle(req))`.

Reply:
I think a helper `handle_if_not_404` closure would be nice.

Reply:
Something like
```rust
let handle_if_404 = |e, handler| if e.response.status == Some(status::NotFound) {
    handler(req)
} else {
    Err(e)
};
```